app/models/manageiq/providers/amazon/network_manager/refresh_parser.rb

```diff
@@ -121,17 +127,39 @@ def get_load_balancer_health_checks
     process_collection(load_balancers, :load_balancer_health_checks) { |lb| parse_load_balancer_health_check(lb) }
   end
 
-
   def get_floating_ips
     ips = @aws_ec2.client.describe_addresses.addresses
+    # Take only floating ips that are not already in stored by ec2 flaoting_ips
+    ips = ips.select { |floating_ip| @data_index.fetch_path(:floating_ips, floating_ip.public_ip).nil? }
     process_collection(ips, :floating_ips) { |ip| parse_floating_ip(ip) }
   end
 
+  def get_public_ips
+    public_ips = []
+    network_ports.each do |network_port|
+      network_port.private_ip_addresses.each do |private_address|
+        if private_address.association && !(public_ip = private_address.association.public_ip).blank?
```

Comment:
yeah, I know :)
I thought this was deprecated in our style guide, but its apparently not.

The thinking behind not using this, is that its easy to slip in a `==`, which could be avoided by [yoda conditions](https://en.wikipedia.org/wiki/Yoda_conditions), but nobody really uses them :)

So, I try to avoid this and rather have an additional and explicit line. And imho this makes the code most of the time more readable.

But in the end its also a matter of taste